src/managers/config.py

```diff
@@ -537,54 +394,50 @@ def server_properties(self) -> List[str]:
             + DEFAULT_CONFIG_OPTIONS.split("\n")
         )
 
-        if self.charm.tls.enabled and self.charm.tls.certificate:
+        if self.state.cluster.tls_enabled and self.state.broker.certificate:
             properties += self.tls_properties + self.zookeeper_tls_properties
 
         return properties
 
-    @staticmethod
-    def _translate_config_key(key):
-        """Format config names into server properties, blacklisted property are commented out.
-
-        Returns:
-            String with Kafka configuration name to be placed in the server.properties file
-        """
-        return key.replace("_", ".") if key not in SERVER_PROPERTIES_BLACKLIST else f"# {key}"
-
     @property
-    def config_properties(self) -> List[str]:
+    def config_properties(self) -> list[str]:
         """Configure server properties from config."""
         return [
             f"{self._translate_config_key(conf_key)}={str(value)}"
-            for conf_key, value in self.charm.config.dict().items()
+            for conf_key, value in self.config.dict().items()
             if value is not None
         ]
 
+    @property
+    def zk_jaas_config(self) -> str:
+        """Builds the JAAS config for Client authentication with ZooKeeper.
+
+        Returns:
+            String of Jaas config for ZooKeeper auth
+        """
+        return f"""
+Client {{
+    org.apache.zookeeper.server.auth.DigestLoginModule required
+    username="{self.state.zookeeper.username}"
+    password="{self.state.zookeeper.password}";
+}};
```

Comment:
**nitpick:** Definitely non-blocking, but I hate that Python makes us do this indent thing.

We could do something like this, but not that important:

```python
return (
    "Client {{"
    "    org.apache.zookeeper.server.auth.DigestLoginModule required"
   f'    username="{self.state.zookeeper.username}"'
   f'    password="{self.state.zookeeper.password}";'
    "}};"
)
```

Reply:
Here without indents python or the linters don't complain. So it's mostly an aesthetics thing. I could also be made a list of strings :man_shrugging: